clarifai/client/model.py

```diff
@@ -1,22 +1,195 @@
     """Initializes an Model object.
     Args:
         model_id (str): The Model ID to interact with.
+        model_version (dict): The Model Version to interact with.
+        output_config (dict): The output config to interact with.
+          min_value (float): The minimum value of the prediction confidence to filter.
+          max_concepts (int): The maximum number of concepts to return.
+          select_concepts (list[Concept]): The concepts to select.
+          sample_ms (int): The number of milliseconds to sample.
         **kwargs: Additional keyword arguments to be passed to the ClarifaiAuthHelper.
     """
-    self.kwargs = {**kwargs, 'id': model_id}
+    self.kwargs = {**kwargs, 'id': model_id, 'model_version': model_version,
+                   'output_info': {'output_config': output_config}}
     self.model_info = resources_pb2.Model(**self.kwargs)
-    super().__init__(user_id=self.user_id, app_id=self.app_id)
+    self.logger = get_logger(logger_level="INFO")
+    BaseClient.__init__(self, user_id=self.user_id, app_id=self.app_id)
+    Lister.__init__(self)
+
+  def predict(self, inputs: List[Input]):
+    """Predicts the model based on the given inputs.
+    Args:
+        inputs (list[Input]): The inputs to predict, must be less than 128.
+    """
+    if len(inputs) > 128:
+      raise UserError("Too many inputs. Max is 128.")  # TODO Use Chunker for inputs len > 128
+
+    request = service_pb2.PostModelOutputsRequest(
+        user_app_id=self.user_app_id,
+        model_id=self.id,
+        version_id=self.model_version.id,
+        inputs=inputs,
+        model=self.model_info)
+
+    start_time = time.time()
+    backoff_iterator = BackoffIterator()
+    while True:
+      response = self._grpc_request(self.STUB.PostModelOutputs, request)
+
+      if response.outputs and \
+        response.outputs[0].status.code == status_code_pb2.MODEL_DEPLOYING and \
+        time.time() - start_time < 60 * 10: # 10 minutes
+        self.logger.info(f"{self.id} model is still deploying, please wait...")
+        time.sleep(next(backoff_iterator))
+        continue
+
+      if response.status.code != status_code_pb2.SUCCESS:
+        raise Exception(f"Model Predict failed with response {response!r}")
+      else:
+        break
+
+    return response
+
+  def predict_by_filepath(self, filepath: str, input_type: str):
+    """Predicts the model based on the given filepath.
+    Args:
+        filepath (str): The filepath to predict.
+        input_type (str): The type of input. Can be 'image', 'text', 'video' or 'audio.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_prediction = model.predict_by_filepath('/path/to/image.jpg', 'image')
+        >>> model_prediction = model.predict_by_filepath('/path/to/text.txt', 'text')
+    """
+    if input_type not in ['image', 'text', 'video', 'audio']:
```

Comment:
It's faster to do set lookups than list lookups. No change required but something you may consider in the future